README.rst

```diff
@@ -0,0 +1,320 @@
+    :target: https://opensource.org/licenses/mit-license.php
+    :alt: MIT License
+
+.. image:: https://badge.fury.io/py/persipubsub.svg
+    :target: https://badge.fury.io/py/persipubsub
+    :alt: PyPI - version
+
+.. image:: https://img.shields.io/pypi/pyversions/persipubsub.svg
+    :alt: PyPI - Python Version
+
+.. image:: https://readthedocs.org/projects/persipubsub/badge/?version=latest
+    :target: https://persipubsub.readthedocs.io/en/latest/?badge=latest
+    :alt: Documentation Status
+
+
+``persipubsub`` implements a persistent, thread-safe and process-safe `lmdb
+<http://www.lmdb.tech/doc/>`_-queue for inter-process communication.
+
+Primarily, we used `zeromq <http://zeromq.org//>`_ for inter-process
+communication with a slight improvement through `persizmq
+<https://github.com/Parquery/persizmq>`_. This still didn't fulfill the level
+of persistence we wanted.
+
+Our motivation was to replace our previous library with a one which is
+similarly easy to setup and to use. Additionally, it should make it possible to
+send `protobuf <https://developers.google.com/protocol-buffers/>`_ messages
+more precisely bytes persistently, thread-safely and process-safely from
+many publishers to many subscribers.
+
+Besides basic publisher and subscriber classes the library offers control
+methods for easy deployment from a config JSON file and maintenance in case
+needed.
+
+Related projects
+================
+
+persist-queue
+-------------
+
+* Offers not all functionality expected from a queue. Has put and get function
+  which are basically only push and pop. So ``front`` functionality is missing.
+  So neither can multiple subscriber be implemented nor can't be guaranteed that
+  no data is lost when a thread fails.
+* All messages in queues are serialized by ``pickle`` which was for us a reason
+  against using this library.
+
+Kafka
+-----
+* For our needs as local system with IPC it's an unnecessary overhead.
+* Integration of Kafka written in Scala and Java in our C++/Python/Go codebase
+  not that simple.
+* Hard to setup and maintain `Kafka <https://kafka.apache.org/>`_.
+* broker system eventually a bottleneck.
+
+RabbitMQ
+--------
+* For our needs as local system with IPC it's an unnecessary overhead.
+* broker system eventually a bottleneck.
+* `RabbitMQ <https://www.rabbitmq.com//>`_ is less scalable than Kafka, but is
+  supported officially by more languages.
+
+zeromq persistence pattern
+--------------------------
+* Only persistence pattern of `zeromq <http://zeromq.org//>`_ is
+  `Titanic <https://rfc.zeromq.org/spec:9/TSP//>`_ which is also a broker
+  system. This takes away the purpose and advantage of zeromq to be a
+  lightweight library which requires no broker.
+
+Usage
+=====
+
+The usage of the library can be differentiated into deployment and actual
+runtime of your processes.
+
+Deployment
+----------
+
+For deployment only a proper config JSON file is needed to setup the whole
```

Comment:
"In the deployment stage the library sets up the queue structure. It is using JSON file blablabla"